simplifier/operations/boolean.py

```diff
@@ -179,7 +182,7 @@ def collapse(self) -> Optional[BitVector]:
         return super().collapse()
 
 
-class Equal(EqualRelation, UnsignedRelation):
+class Equal(EqualRelation, UnsignedRelation, CommutativeOperation):
```

Comment:
I am not sure whether we want to add the class `CommutativeOperation` here. We never call or use any of the functions in `CommutativeOperation` and I also think that some of them would not be correct. 
For example,  `_promote_subexpression` would simplify `(a == (b == 2))` to `a == b == 2`. The condition `(a == (b == 2))` is true if `b =2` and `a=1` or if `b!=2` and `a=0`. But `a == b == 2` is only true if all are `a=b=2`. 

I know, it is a commutative operation. The question is why do we need this here?

Reply:
```suggestion
class Equal(EqualRelation, UnsignedRelation):
```
Ah, I arbitrarily added it as it is a commutative operation. I can remove it for now.